### PR TITLE
Feature/us14148

### PR DESCRIPTION
**Pull request by bwf93**
The new command can be run with:
sudo tools/serviced-storage/serviced-storage sync -p /tmp/outdata/ --source /project/europa/var/serviced/volumes/ -o dm.thinpooldev=/dev/mapper/serviced-serviced--pool

### CODE REVIEW

**Comment by jplouis** [comment]
retest this

**Comment by bwf93** [comment]
retest this since jenkins jobs fail for random reasons

File: tools/serviced-storage/volume.go [hunk 1]
```diff
@@ -41,29 +42,83 @@ type VolumeRemove struct {
 		Name string `description:"Name of the volume to remove"`
 	} `positional-args:"yes" required:"yes"`
 }
+// VolumeSync is the subcommand for syncing two volumes
+type VolumeSync struct {
+	Args struct {
+		DestinationPath flags.Filename `description:"Path of the destionation" long:"destination-path" short:"p" required:"yes" `
+		SourcePath flags.Filename      `description:"Path of the source driver" long:"source" short:"s" required:"yes"`
```
Comment (jafcrocker) [inline tools/serviced-storage/volume.go]: I think these should be positional args, not options.

**Comment by bwf93** [comment]
Changed to use positional arguments for the source and destination and to optionally init the driver.  Also added logging for each volume that is synced.

New example run
sudo tools/serviced-storage/serviced-storage sync /project/europa/var/serviced/volumes/ /tmp/outdata  -c  --type devicemapper -o dm.thinpooldev=/dev/mapper/serviced-serviced--pool

File: tools/serviced-storage/volume.go [hunk 2]
```diff
@@ -44,6 +47,91 @@ type VolumeRemove struct {
...
+			logger.Fatal(err)
+		}
+		logger.Info("Creating driver for destination")
+		initStatus := volume.InitDriver(destinationDriverType, destinationPath, App.Options.Options)
+		if initStatus != nil {
+			logger.Fatal(initStatus)
+		}
+	}
+	destinationDirectory := GetDefaultDriver(destinationPath)
+	logger.Info("Getting driver for destination")
```
Comment (iancmcc) [inline tools/serviced-storage/volume.go]: Make this debug-level. It should nearly never fail.

File: tools/serviced-storage/volume.go [hunk 3]
```diff
@@ -44,6 +47,91 @@ type VolumeRemove struct {
...
+	destinationPath := string(c.Args.DestinationPath)
+	sourcePath := string(c.Args.SourcePath)
+	logger := log.WithFields(log.Fields{
+		"destination":destinationPath,
+		"source":sourcePath})
+	if c.Create {
+		logger = logger.WithFields(log.Fields{
+			"type":c.Type,
+		})
+		logger.Info("Determining driver type for destination")
```
Comment (iancmcc) [inline tools/serviced-storage/volume.go]: Make this debug-level.

File: tools/serviced-storage/volume.go [hunk 4]
```diff
@@ -44,6 +47,91 @@ type VolumeRemove struct {
...
+		}
+	}
+	destinationDirectory := GetDefaultDriver(destinationPath)
+	logger.Info("Getting driver for destination")
+	destinationDriver, err := InitDriverIfExists(destinationDirectory)
+	if err != nil {
+		logger.Fatal(err)
+	}
+	sourceDirectory := GetDefaultDriver(string(sourcePath))
+	logger.Info("Getting driver for source")
```
Comment (iancmcc) [inline tools/serviced-storage/volume.go]: And this.

**Pull request merged**